src/Nethermind/Nethermind.Synchronization/SnapSync/ProgressTracker.cs

```diff
@@ -32,27 +40,72 @@ public class ProgressTracker
+        // throughout the sync. So its easy.
+        private Dictionary<Keccak, AccountRangePartition> AccountRangePartitions { get; set; } = new();
+
+        // Using a queue here to evenly distribute request across partitions. Don't want a situation where one really slow
+        // partition is taking up most of the time at the end of the sync.
+        private ConcurrentQueue<AccountRangePartition> AccountRangeReadyForRequest { get; set; } = new();
         private ConcurrentQueue<StorageRange> NextSlotRange { get; set; } = new();
         private ConcurrentQueue<PathWithAccount> StoragesToRetrieve { get; set; } = new();
         private ConcurrentQueue<Keccak> CodesToRetrieve { get; set; } = new();
         private ConcurrentQueue<AccountWithStorageStartingHash> AccountsToRefresh { get; set; } = new();
 
-        public bool MoreAccountsToRight { get; set; } = true;
 
         private readonly Pivot _pivot;
 
-        public ProgressTracker(IBlockTree blockTree, IDb db, ILogManager logManager)
+        public ProgressTracker(IBlockTree blockTree, IDb db, ILogManager logManager, int accountRangePartitionCount = 8)
         {
             _logger = logManager.GetClassLogger() ?? throw new ArgumentNullException(nameof(logManager));
             _db = db ?? throw new ArgumentNullException(nameof(db));
 
             _pivot = new Pivot(blockTree, logManager);
 
+            _accountRangePartitionCount = accountRangePartitionCount;
+            SetupAccountRangePartition();
+
             //TODO: maybe better to move to a init method instead of the constructor
             GetSyncProgress();
         }
 
+        private void SetupAccountRangePartition()
+        {
+            // Confusingly dividing the range evenly via UInt256 for example, consistently cause root hash mismatch.
+            // The mismatch happens on exactly the same partition every time, suggesting tome kind of boundary issues
+            // either on proof generation or validation.
+            byte curStartingPath = 0;
+            int partitionSize = (256 / _accountRangePartitionCount);
+            if (partitionSize == 0) throw new ArgumentException("Too many snap partition");
+
+            for (var i = 0; i < _accountRangePartitionCount; i++)
+            {
+                AccountRangePartition partition = new AccountRangePartition();
+
+                Keccak startingPath = new Keccak(Keccak.Zero.Bytes.ToArray());
+                startingPath.Bytes[0] = curStartingPath;
+
+                partition.NextAccountPath = startingPath;
+                partition.AccountPathStart = startingPath;
+
+                curStartingPath += (byte)partitionSize;
+
+                Keccak limitPath = new Keccak(Keccak.Zero.Bytes.ToArray());
+                limitPath.Bytes[0] = curStartingPath;
+
+                // Special case for the last partition
+                if (i == _accountRangePartitionCount - 1)
+                {
+                    limitPath = Keccak.MaxValue;
```

Comment:
As this overwrites the Keccak completely, move the `new Keccak` path into `else` statement? As don't need to allocate the `byte[]` for this path?

Reply:
Done